File: 2015/Swift/day01/main.swift
import Foundation

let input = readLine()

var floor = 0
var position = 0
var hasFoundPosition = false

guard let input = input else { throw NSError() }

for character in input {
    if character == "(" {
        floor += 1
    } else if character == ")" {
        floor -= 1
    }

    if !hasFoundPosition {
        position += 1

        if floor == -1 {
            hasFoundPosition = true
        }
    }
}

print(floor)
print(position)

File: 2015/Rust/day01/main.rs
fn main() {
    use std::io::stdin;
    
    let mut floor = 0;
    let mut position = 0;
    let mut has_found_position = false;

    let mut input = String::new();
    stdin().read_line(&mut input).expect("Wrong input");

    for c in input.chars() {
        if c == '(' {
            floor += 1;
        } else if c == ')' {
            floor -= 1;
        }

        if !has_found_position {
            position += 1;

            if floor == -1 {
                has_found_position = true;
            }
        }
    }

    println!("{}", floor);
    println!("{}", position);
}
